component/LinkableText/LinkableText.tsx

```diff
@@ -2,11 +2,11 @@ import Link from "next/link";
 
 const LinkableText = ({text}: {text: string}) => {
     const sanitizeNewLine = text?.split('\n')
-    const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/
+    const regexUrl = /(https?\:\/\/)?([^\.\s]+)?[^\.\s]+\.[^\s]+/gi
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new regular expression `(https?\:\/\/)?([^\.\s]+)?[^\.\s]+\.[^\s]+/gi` seems to be less strict than the previous one. It might match strings that are not valid URLs. For example, it would match "http://..". Consider using a more robust regex for URL validation.
```diff
- const regexUrl = /(https?\:\/\/)?([^\.\s]+)?[^\.\s]+\.[^\s]+/gi
+ const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->